app/validators/blocks/block_validator.py

```diff
@@ -22,6 +25,9 @@ def validate(self):
 
         self.validate_source_references(source_references)
 
+        if self.block["type"] not in ["ListCollector", "PrimaryPersonListCollector"]:
+            self.validate_redirect_to_list_add_block_params()
```

Comment:
We should be checking both sides of this - that the params exist for non-collector blocks, and that they don't exist for collector blocks.